enterprise/cmd/frontend/internal/auth/githuboauth/session.go

```diff
@@ -294,3 +295,55 @@ func (s *sessionIssuerHelper) verifyUserOrgs(ctx context.Context, ghClient *gith
 
 	return false
 }
+
+// verifyUserTeams checks whether the authenticated user belongs to one of the GitHub teams listed in the auth.provider > allowOrgsMap configuration
+func (s *sessionIssuerHelper) verifyUserTeams(ctx context.Context, ghClient *githubsvc.V3Client) bool {
+	var err error
+	hasNextPage := true
+	allowedTeams := make(map[string]map[string]bool, len(s.allowOrgsMap))
+
+	for page := 1; hasNextPage; page++ {
+		var githubTeams []*githubsvc.Team
+
+		githubTeams, hasNextPage, _, err = ghClient.GetAuthenticatedUserTeams(ctx, page)
+		if err != nil {
+			log15.Warn("Could not get GitHub authenticated user teams", "error", err)
+			return false
+		}
+
+		for org, teams := range s.allowOrgsMap {
+			teamsMap := make(map[string]bool)
+			for _, team := range teams {
+				teamsMap[team] = true
+			}
+
+			allowedTeams[org] = teamsMap
+		}
```

Comment:
For now, let's just move construction of this map outside of the loop:
```suggestion
	allowedTeams := make(map[string]map[string]bool, len(s.allowOrgsMap))
	for org, teams := range s.allowOrgsMap {
			teamsMap := make(map[string]bool)
			for _, team := range teams {
				teamsMap[team] = true
			}

			allowedTeams[org] = teamsMap
		}

	for page := 1; hasNextPage; page++ {
		var githubTeams []*githubsvc.Team

		githubTeams, hasNextPage, _, err = ghClient.GetAuthenticatedUserTeams(ctx, page)
		if err != nil {
			log15.Warn("Could not get GitHub authenticated user teams", "error", err)
			return false
		}
```

Reply:
Done, @kopancek.